src/components/Editor/ClientCursor.ts

```diff
@@ -1,104 +1,109 @@
   }
 
-  updateLine(
-    cm: CodeMirror.Editor,
-    fromPos: CodeMirror.Position,
-    toPos: CodeMirror.Position,
-  ) {
+  updateLine(cm: CodeMirror.Editor, fromPos: CodeMirror.Position, toPos: CodeMirror.Position) {
     this.removeLine();
+
     this.lineMarker = cm.getDoc().markText(fromPos, toPos, {
       css: `background-color : ${this.color}; opacity:0.7`,
     });
   }
 
   // when user's cursor hover, show name
-  private showCursorNameReserve(parentEl: Element) {
-    const nameElement = document.createElement('span');
-
-    parentEl.addEventListener('mouseenter', () => {
-      nameElement.textContent = this.id;
-      nameElement.classList.remove('text-remove');
-      nameElement.style.position = 'absolute';
-      nameElement.style.top = `-${this.height}px`;
-      nameElement.style.left = '-2px';
-      nameElement.style.backgroundColor = this.color;
-      nameElement.style.padding = '1px 4px';
-      nameElement.style.borderRadius = '4px 4px 4px 0px';
-      nameElement.style.color = invert(this.color, true);
-
-      parentEl.appendChild(nameElement);
+  private showCursorNameReserve(cursorEl: Element) {
+    const nameEl = document.createElement('span');
+
+    cursorEl.addEventListener('mouseenter', () => {
+      if (this.nameRemoveTimeMap.has(this.id)) {
+        clearTimeout(this.nameRemoveTimeMap.get(this.id)!);
+      }
+
+      nameEl.style.position = 'absolute';
+      nameEl.textContent = this.id;
+      nameEl.style.top = `-${this.height}px`;
+      nameEl.style.left = '-2px';
+      nameEl.style.backgroundColor = this.color;
+      nameEl.style.padding = '1px 4px';
+      nameEl.style.borderRadius = '4px 4px 4px 0px';
+      nameEl.style.color = invert(this.color, true);
+
+      /**
+       * nameEl is being reused.
+       * In order to keep the name visible while the mouse is hovering,
+       * It need to delete the css class containing animation when it is mouseenter and add it when it is mouseleave.
+       */
+      nameEl.classList.remove('text-remove');
+      cursorEl.appendChild(nameEl);
     });
 
-    parentEl.addEventListener('mouseleave', () => {
-      nameElement.className = 'text-remove';
-      nameElement.style.animationDuration = `${duration}s`;
-      nameElement.style.animationDelay = `${delay}s`;
+    cursorEl.addEventListener('mouseleave', () => {
+      nameEl.className = 'text-remove';
+      nameEl.style.animationDuration = `${this.nameAnimationDuration}s`;
+      nameEl.style.animationDelay = `${this.nameAnimationDelay}s`;
 
-      this.removeNameReserve(nameElement);
+      this.removeNameReserve(nameEl);
     });
   }
 
   // After animate, It should actually be deleted it.
-  private removeNameReserve(el: Element) {
-    if (markerRemoveMap.has(this.id)) {
-      clearTimeout(markerRemoveMap.get(this.id));
-    }
-
-    const timeoutId = window.setTimeout(() => {
-      if (el.parentNode) {
-        el.parentNode.removeChild(el);
-        markerRemoveMap.delete(this.id);
-      }
```

Comment:
This logic will be used in the mouseenter event,
L90-92
```
if (markerRemoveMap.has(this.id)) {
    clearTimeout(markerRemoveMap.get(this.id));
}
```


I have excluded the conditions here.
```
if (el.parentNode) {
        el.parentNode.removeChild(el);
        markerRemoveMap.delete(this.id);
}
```